src/mlpack/methods/bayesian_ridge/bayesian_ridge_main.cpp

```diff
@@ -0,0 +1,185 @@
+    " parameter."
+    "\n\n"
+    "The program can also provide predictions for test data using either the "
+    "trained model or the given input model.  Test points can be specified with"
+    " the " + PRINT_PARAM_STRING("test") + " parameter.  Predicted responses "
+    "to the test points can be saved with the " +
+    PRINT_PARAM_STRING("output_predictions") + " output parameter."
+    "\n\n"
+    "For example, the following command trains a model on the data " +
+    PRINT_DATASET("data") + " and responses " + PRINT_DATASET("responses") +
+    "with center set to true and scale set to false (so, Bayesian "
+    "Ridge is being solved, and then the model is saved to " +
+    PRINT_MODEL("bayesian_ridge_model") + ":"
+    "\n\n" +
+    PRINT_CALL("bayesian_ridge", "input", "data", "responses", "responses",
+               "center", 1, "scale", 0, "output_model",
+               "bayesian_ridge_model") +
+    "\n\n"
+    "The following command uses the " + PRINT_MODEL("bayesian_ridge_model") +
+    " to provide predicted responses for the data " + PRINT_DATASET("test") +
+    " and save those responses to " + PRINT_DATASET("test_predictions") + ": "
+    "\n\n" +
+    PRINT_CALL("bayesian_ridge", "input_model", "bayesian_ridge_model", "test",
+               "test", "output_predictions", "test_predictions"));
+
+PARAM_TMATRIX_IN("input", "Matrix of covariates (X).", "i");
+
+PARAM_MATRIX_IN("responses", "Matrix of responses/observations (y).", "r");
+
+PARAM_MODEL_IN(BayesianRidge, "input_model", "Trained BayesianRidge model "
+               "to use.", "m");
+
+PARAM_MODEL_OUT(BayesianRidge, "output_model", "Output BayesianRidge model.",
+                "M");
+
+PARAM_TMATRIX_IN("test", "Matrix containing points to regress on (test "
+                 "points).", "t");
+
+PARAM_TMATRIX_OUT("output_predictions", "If --test_file is specified, this "
+                  "file is where the predicted responses will be saved.", "o");
+
+PARAM_INT_IN("center", "Center the data and fit the intercept. Set to 0 to "
+            "disable",
+             "c",
+             1);
+
+PARAM_INT_IN("scale", "Scale each feature by their standard deviations. "
+             "set to 1 to scale.",
+             "s",
+             0);
+
+static void mlpackMain()
+{
+  int center = CLI::GetParam<int>("center");
+  int scale = CLI::GetParam<int>("scale");
+
+  // Check parameters -- make sure everything given makes sense.
+  RequireOnlyOnePassed({ "input", "input_model" }, true);
+  if (CLI::HasParam("input"))
+  {
+    RequireOnlyOnePassed({ "responses" }, true, "if input data is specified, "
+        "responses must also be specified");
+  }
+  ReportIgnoredParam({{"input", false }}, "responses");
+
+  RequireAtLeastOnePassed({ "output_predictions", "output_model" }, false,
+      "no results will be saved");
+  // Is this line really rigth ? It comes from lars_main.hpp.
+  // ReportIgnoredParam({{ "test", true }}, "output_predictions");
```

Comment:
Yeah, you can add this back in, basically this will report that `output_predictions` is ignored unless `test` is specified.

Reply:
Nice, let's uncomment it.

Reply:
Oh, I guess, does it behave correctly when written like this?

```
ReportIgnoredParam({{ "test", false }}, "output_predictions");
```

I didn't notice the `true` when I looked last time; I think that is backwards. :)